### PR TITLE
Improve offline reproducibility and evaluation diagnostics

### PR DESCRIPTION
**Pull request by Leon-Algo**
### Why
- Make RL evaluation trustworthy (no silent base-model fallback when a wrong checkpoint path is passed).
- Make offline/sandboxed runs easier (repo-local HF datasets cache; robust project/data/output paths).
- Make training loop stable (remove reward-based resampling that can bias training and hang).

### What changed
- Evaluation: strict `--trained_agent_path` validation + explicit logs for “checkpoint requested/loaded” (`sql_rl_gen/generation/evaluate_model.py`).
- Compare-eval helper: generate a Markdown report with run parameters and summary metrics (`base_acc_mean`, `rl_acc_mean`, `diff_count`, `changed_sql_count`) (`scripts/run_compare_eval.py`, `scripts/run_compare_eval.sh`).
- Training loop: remove the “retry until perfect reward” loop; per-step `observe`; clearer progress logging; TensorBoard optional (`sql_rl_gen/generation/rllib/custom_trainer.py`).
- CPU/offline robustness: avoid unconditional `.cuda()`; device detection uses `torch.cuda.is_available()` (`sql_rl_gen/generation/rllib/custom_actor.py`, `sql_rl_gen/generation/envs/utils.py`).
- Data helper: safe extraction script for an existing `spider.zip` (`scripts/setup_spider_data.py`).
- Add `.gitignore` and README notes to keep dataset archives/outputs/caches out of git.

### Data
Official dataset bundle: https://ibm.box.com/v/sql-rl-gen-data  
This PR does not add dataset archives to git. Datasets should be extracted under `data_preprocess/data/`.

### How to test
```bash
conda activate sql-rl-gen
python -m compileall -q sql_rl_gen/generation scripts/*.py configs/*.py data_preprocess/*.py
python data_preprocess/spider_sql_data_process.py --dataset spider

python sql_rl_gen/generation/sql_generation.py \
  --model_name_or_path ./local_models/flan-t5-text2sql-with-schema-v2-unpacked \
  --dataset example_text2sql_spider_train \
  --template llama3 --dataset_name spider \
  --outdir ./output/model_spider_train_smoke_20_pr \
  --steps_n 20 --update_interval 10 --epochs 1 --eval_n_episodes 1

python scripts/run_compare_eval.py \
  --model_name_or_path ./local_models/flan-t5-text2sql-with-schema-v2-unpacked \
  --trained_agent_path ./output/model_spider_train_smoke_20_pr/best \
  --dataset example_text2sql_spider_dev --dataset_name spider --template llama3 \
  --number_of_rows_to_use 200 \
  --out_md output/compare_spider_base_vs_rl_200_pr.md

### CODE REVIEW

**Comment by Leon-Algo** [comment]
Quick validation (offline):
- `python -m compileall -q sql_rl_gen/generation scripts/*.py configs/*.py data_preprocess/*.py`
- `python data_preprocess/spider_sql_data_process.py --dataset spider`
- 20-step smoke train produced: `output/model_spider_train_smoke_20_pr/best` and `output/model_spider_train_smoke_20_pr/20_finish`
- Compare eval (200 rows) report: `output/compare_spider_base_vs_rl_200_pr.md`

Data source (official): https://ibm.box.com/v/sql-rl-gen-data

Note: This PR intentionally does not modify the Eureka reward logic; it focuses on reproducibility, diagnostics, and stability.
